qiskit/circuit/library/arithmetic/adders/qft_adder.py

```diff
@@ -0,0 +1,116 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Compute the sum of two qubit registers using QFT."""
+
+import numpy as np
+
+from qiskit.circuit import QuantumCircuit, QuantumRegister
+from qiskit.circuit.library import QFT
+
+
+class QFTAdder(QuantumCircuit):
+    r"""A circuit that uses QFT to perform in-place addition on two qubit registers.
+
+    Circuit to compute the sum of two qubit registers using modular QFT approach from [1]
+    or non-modular QFT approach from [2].
+    Given two equally sized input registers that store quantum states
+    :math:`|a\rangle` and :math:`|b\rangle`, performs addition of numbers that
+    can be represented by the states, storing the resulting state in-place in the second register:
+
+    .. math::
+
+        |a\rangle |b\rangle \mapsto |a\rangle |(a+b)\ (\mathrm{mod}\ 2^n)\rangle
+
+    Here :math:`|a\rangle` (and correspondingly :math:`|b\rangle`) stands for the direct product
+    :math:`|a_n\rangle \otimes |a_{n-1}\rangle \ldots |a_{1}\rangle \otimes |a_{0}\rangle`
+    which denotes a quantum register prepared with the value :math:`a = 2^{0}a_{0} + 2^{1}a_{1} +
+    \ldots 2^{n}a_{n}` [3]. :math:`|(a+b)\ (mod\ 2^n)\rangle` is the addition result with
+    :math:`(\mathrm{mod}\ 2^n)` indicating that modulo :math:`2^n` addition can be optionally
```

Comment:
Have you checked if the extra backslashes before the whitespaces are valid in RST math mode? Alternatively you could do
```
:math:`|(a + b)\text{ mod } 2^n\rangle`
```

Reply:
I check what the result would be here: https://livesphinx.herokuapp.com. At least according to this online tool it does work like that, but honestly your suggestion looks nicer, so I made a commit with that, thanks :)